src/transformers/trainer.py

```diff
@@ -282,6 +284,13 @@ def __init__(
                 ),
                 FutureWarning,
             )
+        # Some warning for the user related to the usage of datasets that do not implement __len__
+        if args.max_steps > 0:
+            logger.warning("max_steps is given, it will override any value given in num_train_epochs")
```

Comment:
This is documented behavior so this should not be a warning, an info is sufficient.